app/routes/v2/choose-level/index.js

```diff
@@ -21,14 +19,46 @@ module.exports = [{
 {
   method: 'POST',
   path: '/v2/choose-level',
-  handler: async (request, h) => {
-    const applyJourney = await cache.get('apply-journey', request.yar.id)
+  options: {
+    validate: {
+      payload: joi.object({
+        level: joi.any().required()
+      }),
+      failAction: async (request, h, error) => {
+        const { applyJourney, paymentRates } = await getPaymentRates(request, error)
+        if (paymentRates) {
+          return h.view('v2/choose-level/choose-level', ViewModel(
+            applyJourney.selectedSbi.sbi, applyJourney.selectedStandard.name, applyJourney.parcelArea, paymentRates, error
+          )).code(400).takeover()
+        }
+        return h.view('no-response')
+      }
+    },
+    handler: async (request, h) => {
+      const applyJourney = await cache.get('apply-journey', request.yar.id)
 
-    const level = request.payload.level
-    const selectedAmbitionLevel = applyJourney.paymentRates[level]
+      const level = request.payload.level
+      const selectedAmbitionLevel = applyJourney.paymentRates[level]
 
-    await cache.update('apply-journey', request.yar.id, { selectedAmbitionLevel: { name: level, level: selectedAmbitionLevel } })
+      await cache.update('apply-journey', request.yar.id, { selectedAmbitionLevel: { name: level, level: selectedAmbitionLevel } })
 
-    return h.redirect('/v2/summary')
+      return h.redirect('/v2/summary')
+    }
   }
 }]
+
+const getPaymentRates = async (request, error) => {
```

Comment:
This would be be better either as its own Module or as part of the ViewModel. Its just good to keep the routes as clean and light weight as possible